src/xc20pEncryption.ts

```diff
@@ -78,6 +79,84 @@ export function x25519Encrypter(publicKey: Uint8Array, kid?: string): Encrypter
   return { alg, enc: 'XC20P', encrypt, encryptCek }
 }
 
+export type X25519AuthEncryptParams = {
+  kid?: string,
+  skid?: string,
+  apu?: string,
+  apv?: string
+}
+
+export function x25519AuthEncrypter(recipientPublicKey: Uint8Array, senderSecretKey: Uint8Array, 
+  options: Partial<X25519AuthEncryptParams> = {}): Encrypter {
+
+  const alg = 'ECDH-1PU+XC20PKW'
+  const keyLen = 256
+  const crv = 'X25519'
+
+  // It is RECOMMENDED by the ECDH-1PU spec to set apu and apv 
+  // to base64url encoded kid and base64url encoded skid in the
+  // recipient header. For the KDF, if no apu and apv is provided,
+  // it is RECOMMENDED that PartyVInfo and PartyUInfo contain
+  // the base64url decoded skid and kid.
```

Comment:
```suggestion
  // the base64url encoded skid and kid.
```
Should be this?

Reply:
Nope, it is `decoded`. It means, you should decodeBase64url(skid) and decodeBase64url(kid) for the KDF. However, `apu` and `apv` will contain the base64url encoded values.

Reply:
But it raises the question if it is a good idea to expect `apu` and `apv` to be the plain string values rather base64url encoded values in the high level x25519AuthEncrypter API:

```
export function x25519AuthEncrypter(recipientPublicKey: Uint8Array, senderSecretKey: Uint8Array, 
  options: Partial<X25519AuthEncryptParams> = {}): Encrypter
```

I thought it might be more easy for a developer that x25519AuthEncrypter ensure base64url encoding. Thoughts?

Reply:
Don't have a strong sense of how developers  would want to use apu / apv.

Reply:
apu and apv are now expected to be base64url encoded since technically, it could also contain binary data